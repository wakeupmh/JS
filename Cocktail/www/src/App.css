body {
  background-image: linear-gradient(90deg,  #ff9800  35%, #f7f7f7 35%);
}
.App{
  display: flex;
  justify-content: center;
  align-items: center;
}