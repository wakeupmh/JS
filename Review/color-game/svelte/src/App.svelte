<script>
  import { onMount } from 'svelte';
  import ColorContainer from "./components/ColorContainer.svelte";
  import ColorSquare from "./components/ColorSquare.svelte";
  import NavBar from "./components/NavBar.svelte";
  import Title from "./components/Title.svelte";

  $: retrievedColor = null;
  $: result = "";
  $: rgb = null;
  $: newGame = "";
  let colors = [];
  const SQUARES_QUANTITY = 6;

  onMount(() => {
    initGame();
  })
  function handleRestartGame(event) {
    if (event.detail.restartGame) {
      initGame();
      result = "";
      newGame = "";
    }
  }

  function handleClickedColor(event) {
    let clickedColor = event.detail.clickedColor;

    if (clickedColor === retrievedColor) {
      result = "Correct ðŸŽ‰ðŸŽŠ";
      newGame = "Play Again? ðŸ˜Š";
      changeColors(clickedColor);
      h1BgColor = clickedColor;
    } else {
      result = "Try Again ðŸ˜ž";
      setTimeout(() => {
        result = "";
      }, 800);
    }
  }

  function initGame() {
    colors = createRandomColors(SQUARES_QUANTITY);
    retrievedColor = getColor(colors);
    rgb = retrievedColor;
  }

  function getColor(colorsArray) {
    return colorsArray[mathRandom(colorsArray.length)];
  }

  function changeColors(color) {
    colors = new Array(SQUARES_QUANTITY).fill(color);
  }

  function createRandomColors(quantity) {
    let colorsArray = [];
    for (let i = 0; i < quantity; i++) {
      colorsArray.push(randomizeColor());
    }
    return colorsArray;
  }

  function randomizeColor() {
    let r = mathRandom();
    let g = mathRandom();
    let b = mathRandom();
    return `rgb(${r}, ${g}, ${b})`;
  }

  function mathRandom(multiplier = 256) {
    return Math.floor(Math.random() * multiplier);
  }
</script>

<div id="app">
  <Title rgb={rgb} />
  <NavBar
    result={result}
    newGame={newGame}
    on:restartGame={handleRestartGame}
  />
  <ColorContainer>
    {#each colors as color}
      <ColorSquare color={color} on:clickedColor={handleClickedColor} />
    {/each}
  </ColorContainer>
</div>