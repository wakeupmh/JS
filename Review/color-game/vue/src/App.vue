<template>
  <div id="app">
    <ColorContainer>
      <ColorSquare 
        v-for="colors in color"
        :key="color"
      />
    </ColorContainer>
  </div>
</template>

<script>
import ColorContainer from "@/components/ColorContainer";
import ColorSquare from "@/components/ColorSquare";

export default {
  name: "app",
  data: () => ({

  }),
  components: {
    ColorContainer, ColorSquare
  },
  methods: {
    getColor (colorsArray) {
      let randomColor = Math.floor(Math.random() * colorsArray.length);
      return colorsArray[randomColor];
    },
    createRandomColors (quantity) {
      let colorsArray = [];
      for (let i = 0; i < quantity; i++) {
        colorsArray.push(this.randomizeColor());
      }
      return colorsArray;
    },
    randomizeColor () {
      let r = this.mathRandom();
      let g = this.mathRandom();
      let b = this.mathRandom();
      return `rgb(${r}, ${g}, ${b})`;
    },
    mathRandom () {
      return Math.floor(Math.random() * 256);
    }
  },
};
</script>

