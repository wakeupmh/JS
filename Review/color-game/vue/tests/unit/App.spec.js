import App from '@/App'
import { shallowMount } from '@vue/test-utils'

describe('App.test.js',() => {
  let component
  
  beforeEach(() => {
    component = shallowMount(App)
    component.setData({ colors: []});
  })

  // it('should equals to []', () => {
  //   expect(component.vm.color).toEqual(['Dog', 'Cat', 'Fish'])
  // })

  it('has the expected html structure', () => {
    expect(component.element).toMatchSnapshot()
  })
})