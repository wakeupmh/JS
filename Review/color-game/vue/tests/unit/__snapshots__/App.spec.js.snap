// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div
  id="app"
>
  <title-stub
    rgb="rgb(205, 2, 217)"
  />
   
  <navbar-stub />
   
</div>
`;
